lib/promoted/ruby/client/request_builder.rb

```diff
@@ -1,33 +1,95 @@
 module Promoted
   module Ruby
     module Client
-      class LogRequestBuilder
-        attr_reader :delivery_timeout_millis, :session_id,
-                      :metrics_timeout_millis, :should_apply_treatment,
-                      :view_id, :user_id, :insertion, :client_log_timestamp,
+      class RequestBuilder
+        attr_reader   :session_id, :should_apply_treatment,
+                      :view_id, :user_id, :insertion,
                       :request_id, :full_insertion, :use_case, :request, :to_compact_metrics_insertion
 
         def initialize params={}
           @only_log                = params[:only_log] || false
-          @delivery_timeout_millis = params[:delivery_timeout_millis] || DEFAULT_DELIVERY_TIMEOUT_MILLIS
-          @metrics_timeout_millis  = params[:metrics_timeout_millis] || DEFAULT_METRICS_TIMEOUT_MILLIS
           @should_apply_treatment  = params[:should_apply_treatment] || false        
         end
 
         # Populates request parameters from the given arguments, presumed to be a hash of symbols.
         def set_request_params args = {}
           @request                 = args[:request] || {}
           @session_id              = request[:session_id]
-          @user_id                 = request[:user_id]
-          @log_user_id             = request[:log_user_id]
+          @platform_id             = request[:platform_id]
+          @cohort_membership       = request[:cohort_membership]
           @view_id                 = request[:view_id]
           @use_case                = Promoted::Ruby::Client::USE_CASES[request[:use_case]] || 'UNKNOWN_USE_CASE'
           @full_insertion          = args[:full_insertion]
-          @client_log_timestamp    = args[:client_log_timestamp] || Time.now.to_i
           @request_id              = SecureRandom.uuid
+          
+          if request[:user_info]
+            @user_info = request[:user_info]
+            @user_id = @user_info[:user_id]
+            @log_user_id = @user_info[:log_user_id]
+          else
+            @user_id     = request[:user_id]
```

Comment:
I'm happy to whack the two top level parameters which were there before, because they make the code unnecessarily complicated. But I don't think we can raise an error here because userInfo only needs to be set on either the top-level log or child request.

Reply:
Done.